### PR TITLE
Feature/consolidate api endpoints

### PR DESCRIPTION
**Pull request by BrianCorbin**
Soundtrack of this PR: [link to song that really fits the mood of this PR]()

### Motivation

There were lots of duplicate API endpoints and code that could be consolidated

### In this PR
* Removed several API endpoints and consolidated them into 1, most wrt anything in the "Get All" category.
* Updated offset and limit to be u64 instead of string (no reason for it to be string) and added to appropriate places
* Removed default offset/limit

### Future Work
* Add paging results and "next query" in response of paged results

### CODE REVIEW

File: full-service/src/json_rpc/json_rpc_request.rs [hunk 1]
```diff
@@ -157,23 +152,18 @@ pub enum JsonCommandRequest {
     get_account_status {
         account_id: String,
     },
-    get_address_for_account {
-        account_id: String,
-        index: i64,
+    get_accounts {
+        offset: Option<u64>,
```
Comment (Shramp) [inline full-service/src/json_rpc/json_rpc_request.rs:156]: omg yes! I always wondered why offset and limit were strings

File: tools/test.sh [hunk 2]
```diff
@@ -2,12 +2,16 @@
 
 # Copyright (c) 2018-2020 MobileCoin Inc.
 
+# RUSTFLAGS="-C instrument-coverage" \
```
Comment (Shramp) [inline tools/test.sh:5]: delete comment

**Review by Shramp — APPROVED** [review]
nice! lgtm

**Comment by sugargoat** [comment]
What's the migration plan for the deprecated APIs?

File: full-service/src/json_rpc/json_rpc_request.rs (hunk 1 above)
Comment (BrianCorbin) [reply full-service/src/json_rpc/json_rpc_request.rs:156]: same!

**Review by Shramp — APPROVED** [review]
lgtm

File: tools/test.sh (hunk 2 above)
Comment (BrianCorbin) [reply tools/test.sh:5]: gonna leave that in for reference in the future

File: full-service/src/db/account.rs [hunk 3]
```diff
@@ -159,7 +159,11 @@ pub trait AccountModel {
     ///
     /// Returns:
     /// * Vector of all Accounts in the DB
-    fn list_all(conn: &Conn) -> Result<Vec<Account>, WalletDbError>;
+    fn list_all(
+        conn: &Conn,
+        offset: Option<u64>,
```
[the remark this answers is not in the document]
Comment (BrianCorbin) [reply full-service/src/db/account.rs:164]: I think it makes more sense to leave these as separate params for now

File: full-service/src/json_rpc/e2e_tests/transaction/build_submit/build_then_submit.rs [hunk 4]
```diff
@@ -338,7 +338,11 @@ mod e2e_transaction {
         let body = json!({
             "jsonrpc": "2.0",
             "id": 1,
-            "method": "get_all_transaction_logs_ordered_by_block",
+            "method": "get_transaction_logs",
+            "params": {
+                "min_block_index": "14",
+                "max_block_index": "14",
```
[the remark this answers is not in the document]
Comment (BrianCorbin) [reply full-service/src/json_rpc/e2e_tests/transaction/build_submit/build_then_submit.rs:327]: maybe in a future PR, as that has to do with jsonRPC not being able to handle u64 values so we have to pass them as strings and convert

**Pull request merged**